### DISCUSSION TITLE
How should we orgnize Jdbc connectors?

### DISCUSSION THREAD

**Discussion by FANNG1** [apache/gravitino#651] [General]
1. use a  separate connector,  like `mysql-connector`,  `pg-connector`
2. use a common connector, like `jdbc-connector` with different types like `mysql`, `pg`

**Comment by FANNG1** [comment]
I prefer using a separate connector.  because:
1. It's straight-forward to end users,    if using `jdbc-connector`,   `provider` and `type` identifies a real connector.   this produces complex for users and other modules
2. the main benefit of `jdbc-connector` is code-reuse,  different types just provide `TypeTransform` and `ExceptionMapper`.  If a engine 100% compatible with MySQL, we could just reuse  MySQL connector without adding a new connector.  I think this condition is rare,  every engine has it's own features (column types, table properteis),  even if a engine 100% compatible with MySQL, we could simple add a xx-connector  which extends JdbcConnector with existing `MysqlTypeTransform` and `MysqlExecptionMapper`,  it doesn't take much work.
3.  Trino, Metacat use separate Jdbc connectors too.  if we collaborate with Trino , it's more straight forward to provide a separate jdbc connector.

**Comment by FANNG1** [comment]
@jerryshao @xunliu @Clearvive @mchades @yuqi1129  what's your advise?

**Comment by diqiu50** [comment]
I agree case 1.  
The JDBC connector is an abstract concept. It has specific implementations like MySQL connector, PostgreSQL connector, and so on. For users, it's easier to understand something like MySQL connector than just a generic JDBC connector. Products like DataGrid, Nativecat, DBeaver, Trino, etc., also enable users to directly use specific implementations like MySQL connector.
